active/0006-extensibility-support-for-deployment-stacks.md

```diff
@@ -0,0 +1,998 @@
+  }
+}
+```
+
+ARM template:
+```json5
+{
+  "extensionConfigs": {
+    "k8s": {
+      "str": "[createObject('value', parameters('strParam'))]",
+      "obj": "[createObject('value', parameters('objParam'))]",
+      "secureStr": "[createObject('value, parameters('secureStrParam'))]",
+      "secureStrExt": "[extensionConfigs('k8s').kubeConfig]",
+      "secureObjExt": "[extensionConfigs('k8s').kubeObject]",
+      "ifBothExt": "[if(parameters('boolParam'), extensionConfigs('k8s').kubeConfig, extensionConfigs('k8s2').kubeConfig)]",
+      "ifMixed": "[if(parameters('boolParam'), extensionConfigs('k8s').kubeConfig, createObject('value', parameters('strParam')))]"
+    }
+  }
+}
+```
+
+### Microsoft.Resources/deployments API changes
+
+To support stacks extensibility, the deployments API needs to return additional data about extensible resources.
+It needs to return extensions with their configurations and the resources deployed to those extensions. Extensible resources
+need to be linked to extensions. This can be done with the key: extension name, extension version, and config ID. It
+is possible the key may match to multiple extensions with differing config instructions, but this is an edge case.
+
+Deployment GET:
+```json5
+{
+  // ... deployment contract
+  "extensions": [ // NEW: "Extensions" must be returned for any deployment terminal state (Succeeded, Failed, etc)
+    {
+      "name": "Kubernetes",
+      "version": "1.0.0",
+      "configId": "<ID from extension>", // Provided by the extension in the CREATE/UPDATE return payload.
+      "config": {
+        // the language expressions provided to the PUT have been evaluated
+        "namespace": {
+          "type": "string",
+          "value": "default"
+        },
+        "kubeConfig": {
+          "type": "securestring",
+          // or keyVaultReference
+          "apiReference": {
+            "method": "POST",
+            "armResourceId": "/subscriptions/.../resourceGroups/.../Microsoft.ContainerService/managedClusters/...",
+            "apiVersion": "2024-02-01",
+            "action": "listClusterAdminCredentials",
+            "query": "a=1&b=2",
+            "responseValuePath": "kubeconfigs[0].value"
+          }
+        }
+      }
+    }
+  ],
+  "outputResources": [ // "outputResources" is only populated when the deployment is successful. Stacks uses operations to peel through failed deployment resources.
+    { // NEW: Extensible resource
+      "extensionName": "Kubernetes",
+      "extensionVersion": "1.0.0",
+      "extensionConfigId": "<ID from extension>",
+      "extensibleResourceType": "core/Service",
+      "apiVersion": "v1",
+      "identifiers": {
+        "metadata": {
+          "namespace": "default",
+          "name": "myService",
+        }
+      }
+    },
+    { // ARM resource
+      "id": "..."
+    }
+  ]
+}
+```
+
+Operation GET for extensible resource:
+```diff
+{
+  ...
+  "type": "Microsoft.Resources/deployments",
```

Comment:
[nit] is this the correct resource type?

Reply:
Removed.